app/src/main/java/sw19/moring03/paint/MainActivity.java

```diff
@@ -139,4 +156,31 @@ public void chooseNewColor(View view) {
 
         colorChooserMenu.dismiss();
     }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == CAMERA_REQUEST) {
+            try {
+                Bitmap cameraPicture = (Bitmap) data.getExtras().get("data");
+
+                if (cameraPicture == null) {
+                    Toast toast = Toast.makeText(getApplicationContext(),
+                            "Could not load image from Camera!", Toast.LENGTH_SHORT);
+                    toast.show();
+                    return;
+                }
+                newPhoto = cameraPicture;
+                setChosenTool(Tool.TAKE_PHOTO);
```

Comment:
pls have a look at the newset feature/PAINT-018 there i've fixed the problem that you have to click an aditional time to load the picture itself. If you can't find it message me or ask me on wednesday and we can have a look at it together